docs/manual/src/internals/ffi_converter_traits.md

```diff
@@ -0,0 +1,68 @@
+Each trait handles a single step in the lifting/lowering process (e.g. lifting an argument, lowering a return, etc.).
+We implement the trait for each type used in the exported API then use that trait in the codegen.
+
+For example, `uniffi::Lift` is used to lift values.
+To handle a function like `fn print(msg: String)`, the generated code will use:
+
+* `<String as Lift<crate::UniFfiTag>>::FfiType` when it needs to specify the FFI type (`RustBuffer` for strings).
+* `<String as Lift<crate::UniFfiTag>>::try_lift()` when it needs to lift an argument value.  In this example, this means taking an FFI value (`<String as Lift<crate::UniFfiTag>>::FfiType` AKA `RustBuffer`) and converting it into a Rust String for passing to the Rust function.
+
+Using a trait for this is important for proc-macros, which only see Rust tokens and don't know the surrounding context.
+For example, if macros always used `RustBuffer` as the FFI type whenever it sees "String", then that would fail if users created a type alias like `type MyTypeAlias = String`.
+This may be unusual for `String`, but it's very common for `Result`.
+In general, any reasoning about the tokens is fragile and should be avoided.
+
+### `UniFfiTag` and the orphan rule
+
+One odd part about the above code is that the `Lift` trait has a generic parameter which is always set to `crate::UniFfiTag`.
+In general, all of the FFI converter traits have this parameter (i.e. we generate `Lower<crate::UniFfiTag>`, `LowerReturn<crate::UniFfiTag>`, etc.).
+What's the point of all of this?
+
+The main reason is to work around issues with the Rust orphan rule and types from 3rd-party crates.
+For example, the [custom types](../types/custom_types.md) documentation shows how `url::Url` can be used in an exported API.
+For these types, we normally can't implement `Lift` in the code we generate in the crate since neither `uniffi::Lift` or `url::Url` is local to that crate.
+This same issue applies to all of the FFI converter traits.
+
+To work around this we:
+
+* Add a generic parameter to each trait (`Lift` becomes `Lift<UT>` where "UT" is short for `UniFfiTag`).
+* Define a unit struct in each crate named `UniFfiTag` (the term "tag" is borrowed from the [C++ template pattern](https://www.geeksforgeeks.org/tag-dispatch-in-cpp/)).
+* We use that unit struct as the generic parameter for the trait (e.g. `Lift<crate::UniFfiTag>` is used to lift a value).
+
+Using the local type as a generic parameter means the impl no longer violates the orphan rule.
+For details on this see the [Rust Chalk Book](https://rust-lang.github.io/chalk/book/clauses/coherence.html#the-orphan-rules-in-rustc)
+The TLDR is that generic parameters "count" towards the requirement that there be a local type in the impl.
+
+In essence, `Lift<crate::UniFfiTag>` is a crate-local version of the `Lift` trait which gets around the orphan rule.
```

Comment:
These two sentences seem to be roughly the same thing. IMO the first one can be removed.